packages/flutter_tools/test/general.shard/build_runner/multiroot_asset_reader_test.dart

```diff
@@ -34,17 +37,27 @@ void main() {
     });
 
     test('Can find assets from the generated directory', () => testbed.run(() async {
-      final MultirootFileBasedAssetReader reader = MultirootFileBasedAssetReader(
-        packageGraph,
-        fs.directory(fs.path.join('.dart_tool', 'build', 'generated'))
-      );
-
-      // Note: we can't read from the regular directory because the default
-      // asset reader uses the regular file system.
-      expect(await reader.canRead(AssetId('foobar', 'lib/bar.dart')), true);
-      expect(await reader.readAsString(AssetId('foobar', 'lib/bar.dart')), 'bar');
-      expect(await reader.readAsBytes(AssetId('foobar', 'lib/bar.dart')), utf8.encode('bar'));
-    }));
+      await IOOverrides.runWithIOOverrides(() async {
+        final MultirootFileBasedAssetReader reader = MultirootFileBasedAssetReader(
+          packageGraph,
+          fs.directory(fs.path.join('.dart_tool', 'build', 'generated'))
+        );
+        expect(await reader.canRead(AssetId('foobar', 'lib/bar.dart')), true);
+        expect(await reader.canRead(AssetId('foobar', 'lib/main.dart')), true);
+
+        expect(await reader.readAsString(AssetId('foobar', 'lib/bar.dart')), 'bar');
+        expect(await reader.readAsString(AssetId('foobar', 'lib/main.dart')), 'main');
+
+        expect(await reader.readAsBytes(AssetId('foobar', 'lib/bar.dart')), utf8.encode('bar'));
+        expect(await reader.readAsBytes(AssetId('foobar', 'lib/main.dart')), utf8.encode('main'));
+
+        expect(await reader.findAssets(Glob('**')).toList(), unorderedEquals(<AssetId>[
+          AssetId('foobar', 'pubspec.yaml'),
+          AssetId('foobar', 'lib/bar.dart'),
+          AssetId('foobar', 'lib/main.dart'),
+        ]));
+      }, FlutterIOOverrides(fileSystem: fs));
+    }), skip: Platform.isWindows);
```

Comment:
Can you give an example of the normalization that is expected, or how this fails on Windows? We should try to track this down to avoid similar problems later.

Reply:
Added a comment describing the issue